### PR TITLE
feat(api): allow to use podman REST API for attaching streams of containers

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Allow to use the dedicated Podman API for attaching stream of containers

### What issues does this PR fix or reference?

need for https://github.com/containers/podman-desktop/issues/970

### How to test this PR?

Unit test provided

### CODE REVIEW

File: packages/main/src/plugin/dockerode/libpod-dockerode.ts [hunk 1]
```diff
@@ -206,6 +207,44 @@ export class LibpodDockerode {
       });
     };
 
+    // add attach
+    prototypeOfDockerode.podmanAttach = function (containerId: string) {
```
Comment (dgolovin) [inline packages/main/src/plugin/dockerode/libpod-dockerode.ts:211]:
```suggestion
    prototypeOfDockerode.dial = function (optsf: any) {
      return new Promise((resolve, reject) => {
        this.modem.dial(optsf, (err: unknown, data: unknown) => {
          if (err) {
            return reject(err);
          }
          resolve(data);
        });
      });
    }
    // add attach
    prototypeOfDockerode.podmanAttach = function (containerId: string) {
```

File: packages/main/src/plugin/dockerode/libpod-dockerode.ts [hunk 2]
```diff
@@ -206,6 +207,44 @@ export class LibpodDockerode {
...
+      };
+
+      return new Promise((resolve, reject) => {
+        this.modem.dial(optsf, (err: unknown, stream: unknown) => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(stream);
+        });
+      });
```
Comment (dgolovin) [inline packages/main/src/plugin/dockerode/libpod-dockerode.ts:245]:
```suggestion
      return dial(optsf);
```

Comment (dgolovin) [reply packages/main/src/plugin/dockerode/libpod-dockerode.ts:245]: and down below where applicable to avoid code duplication.

**Review by dgolovin — APPROVED** [review]
LGTM with one optional change to reduce code dup.

File: packages/main/src/plugin/dockerode/libpod-dockerode.ts (hunk 2 above)
Comment (benoitf) [reply packages/main/src/plugin/dockerode/libpod-dockerode.ts:245]: it looks like a refactoring as it applies to all the other methods, so I'll handle in a follow-up PR.

for now it was reusing the same approach than the dockerode library 
 https://github.com/apocas/dockerode/blob/master/lib/docker.js with duplication.

**Pull request merged**